src/commands/createStaticWebApp/TemplateListStep.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep, IAzureQuickPickItem } from 'vscode-azureextensionui';
+import { RepoData } from '../../gitHubTypings';
+import { getTemplateRepos } from '../../utils/gitHubUtils';
+import { localize } from '../../utils/localize';
+import { nonNullProp } from '../../utils/nonNull';
+import { IStaticWebAppWizardContext } from './IStaticWebAppWizardContext';
+
+export class TemplateListStep extends AzureWizardPromptStep<IStaticWebAppWizardContext> {
+    public async prompt(context: IStaticWebAppWizardContext): Promise<void> {
```

Comment:
To make it cache template entries, you can do something like this:

```
private _templateRepos: RepoData[] | undefined;
    public async prompt(context: IStaticWebAppWizardContext): Promise<void> {

        const placeHolder: string = localize('chooseTemplatePrompt', 'Choose a template for the new static web app.');

        const getPicks = async () => {
            this._templateRepos ||= await getTemplateRepos(context);
            return this._templateRepos.map((repo: RepoData) => ({ label: repo.name, data: repo }));
        }
```

That way, you wont' have to add more things to the context.